core/src/test/scala/scala/pickling/tags/TypeTagsTest.scala

```diff
@@ -0,0 +1,35 @@
+package scala.pickling.tags
+
+import org.scalatest.FunSuite
+import scala.reflect.runtime.{universe => ru, currentMirror}
+
+ case class NestedCaseClass(x: String)
+
+class TagTest extends FunSuite {
+ 
+  test("makeRaw is consistent") {
+    val tags = List(
+      FastTypeTag[List[_]],
+      FastTypeTag[Int],
+      FastTypeTag[String],
+      FastTypeTag[NestedCaseClass],
+      FastTypeTag[NestedCaseClass.type],
+      FastTypeTag[List.type],
+      FastTypeTag[java.util.List[_]],
+      FastTypeTag[Option[_]],
+      FastTypeTag[Some[_]],
+      // TODO - Figure out why these are broken.
+      FastTypeTag[String with Serializable],
+      FastTypeTag[Any => Any]
```

Comment:
Some other exotic types to consider. Type projections. Annotated types. Existential types with restrictions.